jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js

```diff
@@ -124,39 +347,73 @@ YUI.add('deployment-payment', function() {
               Name and address
             </h2>
             {this._generateBusinessNameField()}
-            {this._generateAddressFields()}
+            <juju.components.GenericInput
+              disabled={disabled}
+              label="Email address"
+              ref="emailAddress"
+              required={true}
+              validate={[required]} />
+            {this._generateAddressFields('user')}
             <h2 className="deployment-payment__title">
               Payment information
             </h2>
             <juju.components.GenericInput
               disabled={disabled}
               label="Card number"
-              required={true} />
-            <div className="twelve-col">
-              <div className="six-col">
+              onChange={this._formatCardNumber}
+              ref="cardNumber"
+              required={true}
+              validate={[required, {
+                regex: /^[a-zA-Z0-9_-\s]{16,}/,
+                error: 'The card number is too short.'
+              }, {
+                regex: /^[a-zA-Z0-9_-\s]{0,23}$/,
+                error: 'The card number is too long.'
+              }, {
+                regex: /^[0-9\s]+$/,
+                error: 'The card number can only contain numbers.'
+              }]} />
+            <div className="twelve-col no-margin-bottom">
+              <div className="six-col no-margin-bottom">
                 <juju.components.GenericInput
                   disabled={disabled}
                   label="Expiry MM/YY"
-                  required={true} />
+                  ref="cardExpiry"
+                  required={true}
+                  validate={[required, {
+                    regex: /[\d]{2}\/[\d]{2}/,
+                    error: 'The expiry must be in the format MM/YY'
```

Comment:
I suppose this is fine, I've typically also seen MMYY without any divisor and then put a 4 character limit on the field.